symbolic-debuginfo/src/dwarf.rs

```diff
@@ -51,29 +52,76 @@ fn offset(addr: u64, offset: i64) -> u64 {
     (addr as i64).wrapping_sub(offset as i64) as u64
 }
 
-/// An error handling [`DWARF`](trait.Dwarf.html) debugging information.
+/// The error type for [`DwarfError`].
 #[non_exhaustive]
-#[derive(Debug, Error)]
-pub enum DwarfError {
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum DwarfErrorKind {
     /// A compilation unit referenced by index does not exist.
-    #[error("compilation unit for offset {0} does not exist")]
     InvalidUnitRef(usize),
 
     /// A file record referenced by index does not exist.
-    #[error("referenced file {0} does not exist")]
     InvalidFileRef(u64),
 
     /// An inline record was encountered without an inlining parent.
-    #[error("unexpected inline function without parent")]
     UnexpectedInline,
 
     /// The debug_ranges of a function are invalid.
-    #[error("function with inverted address range")]
     InvertedFunctionRange,
 
     /// The DWARF file is corrupted. See the cause for more information.
-    #[error("corrupted dwarf debug data")]
-    CorruptedData(#[from] GimliError),
```

Comment:
`GimliError` was exposed because of the pub usage here. Maybe we can hide all the `gimli` re-exports again?

Reply:
Good catch, I'll create a follow-up.

Reply:
https://github.com/getsentry/symbolic/pull/300